src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -780,28 +781,58 @@ impl<T: AstInfo> AstDisplay for CreateTypeStatement<T> {
         f.write_str("CREATE TYPE ");
         f.write_node(&self.name);
         f.write_str(" AS ");
-        f.write_str(&self.as_type);
-        f.write_str("( ");
-        if !self.with_options.is_empty() {
-            f.write_node(&display::comma_separated(&self.with_options));
-        }
-        f.write_str(" )");
+        match self.as_type {
+            CreateTypeAs::List | CreateTypeAs::Map => {
+                f.write_str(&self.as_type);
+                f.write_str("( ");
+                if !self.with_options.is_empty() {
+                    f.write_node(&display::comma_separated(&self.with_options));
+                }
+                f.write_str(" )");
+            }
+            CreateTypeAs::Record => {
+                f.write_str("( ");
+
+                // TODO(phemberger): hmm do we create a separate field to store ColumnDefs directly?
```

Comment:
Yeah, 100% we do. Seems to me like `CreateTypeAs` wants to look something like this:

```rust
enum CreateTypeAs {
    List { with_options: Vec<SqlOption<T>> },
    Map { with_options: Vec<SqlOption<T>> },
    Record { column_defs: Vec<ColumnDef<T>> }
}
```

Reply:
Nice, that's what I've been leaning towards too. I'm a little scared that it'll cause a wide impact (I think there are several spots where it's helpful the Record column defs are treated as SqlOptions), so fingers crossed!

Reply:
phew, wasn't bad and I think this helped tidy up the overall flow of things